tests/test_cost_registry.py

```
@@ -58,4 +65,76 @@ def test_unknown_ticker():
+    ):
+        spec = resolve_symbol_onchain(UNKNOWN, w3, 1)
+    assert spec.symbol == UNKNOWN
+
+
+def test_resolve_symbol_onchain_no_pool():
+    w3 = MagicMock()
+    with patch(
+        "orion_finance_sdk_py.costs.registry.factory_get_pool",
+        return_value=ZERO_ADDRESS,
+    ):
+        with pytest.raises(KeyError, match="No Uniswap v3 USDC pool"):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)